assignment-2/app/src/main/java/com/example/assignment2/MainActivity.kt

```diff
@@ -1,11 +1,75 @@
 package com.example.assignment2
 
+import android.annotation.SuppressLint
+import android.graphics.Color
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.widget.TextView
+import androidx.activity.viewModels
+import androidx.core.view.forEachIndexed
+import com.example.assignment2.databinding.ActivityMainBinding
 
 class MainActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityMainBinding
+    // MyViewModel class의 instance인 viewModel을 만들고, 'by viewModels()'로 생성을 위임한다. MyViewModel.kt가 MainActivity.kt와 같은 package에 있어서 import 안 해도 된다.
+    private val viewModel: MyViewModel by viewModels()
+    @SuppressLint("NotifyDataSetChanged") // 초기화 되거나 되돌아가기 버튼을 사용할 때에 대비하여 그냥 전체 데이터가 바뀌었는지를 알려주는 함수를 사용함.
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        ActivityMainBinding.inflate(layoutInflater).also { this.binding = it }
+        setContentView(binding.root)
+
+        // Click이 들어오면 GameState를 바꿈 - 유효한 클릭인지에 따라 바뀌지 않을 수도 있음
+        binding.button0.setOnClickListener { viewModel.changeGameState(0, 0) }
+        binding.button1.setOnClickListener { viewModel.changeGameState(0, 1) }
+        binding.button2.setOnClickListener { viewModel.changeGameState(0, 2) }
+        binding.button3.setOnClickListener { viewModel.changeGameState(1, 0) }
+        binding.button4.setOnClickListener { viewModel.changeGameState(1, 1) }
+        binding.button5.setOnClickListener { viewModel.changeGameState(1, 2) }
+        binding.button6.setOnClickListener { viewModel.changeGameState(2, 0) }
+        binding.button7.setOnClickListener { viewModel.changeGameState(2, 1) }
+        binding.button8.setOnClickListener { viewModel.changeGameState(2, 2) }
+        binding.reset.setOnClickListener {
+            viewModel.backTo(0)
+            binding.reset.apply{
+                text = "초기화"
+                setBackgroundColor(Color.parseColor("#C2BEC0"))
+            }
+        }
+
+        // drawer image가 클릭되면 drawer 열기
+        binding.drawerIcon.setOnClickListener { binding.root.openDrawer(binding.drawer) }
+
+        // RecyclerView에 adpater 연결
+        binding.recyclerView.adapter = MyMultiAdapter( viewModel.history, { turn: Int -> viewModel.backTo(turn) } )
```

Comment:
[nit] 마지막 파라미터가 람다인 경우 
```suggestion
        binding.recyclerView.adapter = MyMultiAdapter( viewModel.history) { turn -> 
            viewModel.backTo(turn) 
        } 
```
이렇게 주로 씁니다. 그 람다가 (예를 들어) onBack인지 onClick인지 명시해 주고 싶을 때는 () 안에 {} 를 넣기도 하는데, 그럴 때는 파라미터 이름도 명시해 주죠
```
        binding.recyclerView.adapter = MyMultiAdapter( viewModel.history, goBack = { turn ->
            viewModel.backTo(turn)
        })
```
이렇게요